src/sqlancer/postgres/PostgresProvider.java

```diff
@@ -218,24 +220,51 @@ protected TestOracle getTestOracle(PostgresGlobalState globalState) throws SQLEx
 
     @Override
     public Connection createDatabase(PostgresGlobalState globalState) throws SQLException {
-        String url = "jdbc:postgresql://localhost:5432/test";
+        String username = globalState.getOptions().getUserName();
+        String password = globalState.getOptions().getPassword();
+        String entryPath = "test";
+        String entryURL = globalState.getDmbsSpecificOptions().connectionURL;
+        // trim URL to exclude "jdbc:"
+        if (entryURL.startsWith("jdbc:")) {
+            entryURL = entryURL.substring(5);
+        }
+        try {
+            URI uri = new URI(entryURL);
+            String userInfoURI = uri.getUserInfo();
+            String pathURI = uri.getPath();
+            if (userInfoURI != null) {
+                // username and password specified in URL take precedence
+                String[] userInfo = userInfoURI.split(":", 2);
+                username = userInfo[0];
+                password = userInfo[1];
+                int userInfoIndex = entryURL.indexOf(userInfoURI);
+                entryURL = entryURL.substring(0, userInfoIndex)
+                        + entryURL.substring(userInfoIndex + userInfoURI.length() + 1);
+            }
+            if (pathURI != null) {
+                entryPath = pathURI;
+            }
+        } catch (URISyntaxException e) {
+            throw new AssertionError(e);
+        }
         String databaseName = globalState.getDatabaseName();
-        Connection con = DriverManager.getConnection(url, globalState.getOptions().getUserName(),
-                globalState.getOptions().getPassword());
-        globalState.getState().logStatement("\\c test;");
+        Connection con = DriverManager.getConnection("jdbc:" + entryURL, username, password);
+        globalState.getState().logStatement(entryURL);
         globalState.getState().logStatement("DROP DATABASE IF EXISTS " + databaseName);
         String createDatabaseCommand = getCreateDatabaseCommand(databaseName, con, globalState);
         globalState.getState().logStatement(createDatabaseCommand);
-        globalState.getState().logStatement("\\c " + databaseName);
         try (Statement s = con.createStatement()) {
             s.execute("DROP DATABASE IF EXISTS " + databaseName);
         }
         try (Statement s = con.createStatement()) {
             s.execute(createDatabaseCommand);
         }
         con.close();
-        con = DriverManager.getConnection("jdbc:postgresql://localhost:5432/" + databaseName,
-                globalState.getOptions().getUserName(), globalState.getOptions().getPassword());
+        int pathIndex = entryURL.indexOf(entryPath);
+        String testURL = entryURL.substring(0, pathIndex + 1) + databaseName
```

Comment:
Same here: I think it would be cleaner to define the variables once, and then use these, rather than using the `substring` method.

Reply:
The only issue is that these are a combination of many possible things that the user can decide to include in the URL. This makes it hard to define variables, but allows more flexibility in setting optional parameters.

Reply:
Okay, I see. Thanks for the explanation! Maybe as a future PR, we could consider adding test cases to check if connection strings are correctly parsed (although it doesn't have a high priority at the moment).